File: frontend/board/src/app/modules/kanban/models/sprint-user-settings.ts
export type SprintUserSettingsUser = {
  id: number
  name: string
  avatar_url: string
}

export type SprintUserSettings = {
  id: number
  sprint_id: number
  user_id: number
  user: SprintUserSettingsUser
  hours_per_user: number
}

export type SaveSprintUserSettingsRequest = {
  user_id: number
  hours_per_user: number
}

File: backend/internal/infra/persistance/models/sprint_user_settings.go
package models

import "time"

type SprintUserSettings struct {
	CreatedAt    time.Time
	UpdatedAt    time.Time
	Id           uint `gorm:"id;primaryKey"`
	SprintId     uint `gorm:"sprint_id;not null;uniqueIndex:sprint_user_idx"`
	UserId       uint `gorm:"user_id;not null;uniqueIndex:sprint_user_idx"`
	User         User `gorm:"foreignKey:UserId"`
	HoursPerUser uint `gorm:"hours_per_user;default:0;not null"`
}
